include/libdnf/utils/exception.hpp

```diff
@@ -79,47 +71,19 @@ class SystemError : public RuntimeError {
 public:
     SystemError(int error_code, const std::string & what) : RuntimeError(what), error_code{error_code} {}
     SystemError(int error_code, const char * what) : RuntimeError(what), error_code{error_code} {}
-    SystemError(int error_code, const std::string & what, const std::string & path) : RuntimeError(what), error_code{error_code}, path{path} {}
     const char * get_domain_name() const noexcept override { return "system"; }
-    const char * get_name() const noexcept override { return "SystemError"; }
-    const char * get_description() const noexcept override { return "System error"; }
+    const char * get_name() const noexcept override;
+    const char * get_description() const noexcept override;
+
+    /// Returns original error code
     int get_error_code() const noexcept { return error_code; }
-    const std::string & get_path() const noexcept { return path; }
-    const char * what() const noexcept override {
-        try {
-            // example: "General RuntimeError exception: [Errno 1] "
-            str_what = get_description();
-            str_what += ": [Errno ";
-            str_what += std::to_string(get_error_code());
-            str_what += "] ";
-
-            // example: "Operation not permitted"
-            str_what += std::system_category().default_error_condition(error_code).message();
-
-            // example: ": Unable to install package"
-            if (runtime_error::what()) {
-                str_what += ": ";
-                str_what += runtime_error::what();
-            }
-
-            // example: ": /path/to/file"
-            if (!path.empty()) {
-                str_what += ": ";
-                str_what += get_path();
-            }
-
-            return str_what.c_str();
-        } catch (...) {
-            return Exception::what();
-        }
-    }
 
 private:
-    const int error_code;
-    const std::string path;
+    int error_code;
+    mutable char name[20] = "Errno_";
```

Comment:
I am not sure whether the format is an optimal. It will generate `Errno_2`. What about to use a space => `Errno 2`?

Reply:
Agreed with @j-mracek but I'd got a step or two further to not add (from an unassuming not that technical user point of view) cryptic syntax and aim for a nice human-readable text (adding namespaces and :: and class names etc. doesn't add any value IMO):
`System Error: 2 - No such file or directory: nonexistent.file`

This assumes everyone knows the number is an `errno` (and if the reader doesn't know what errno is, adding "errno" to the message won't help them much), but if we want to have the "errno" in there:
`System Error: errno 2 - No such file or directory: nonexistent.file`

Just keep it as simple as possible.